### PR TITLE
feat():canal-adapter同步到rdb时，mirrorDb模式下支持改库名

### PR DESCRIPTION
**Pull request by sgjr**
- 使用场景为：MySQL同步数据到MySQL，下游MySQL实例中的库名需要修改， 相关issue: https://github.com/alibaba/canal/issues/5216
- 当 dbMapping 中 mirrorDb 为 true时，支持通过设置 targetDb 来进行改库名操作
- rdb配置实例如下

```
concurrent: true
dataSourceKey: demoSyncDS
dbMapping:
  commitBatch: 3000
  database: test0718
  mirrorDb: true
  targetDb: test0723
destination: demo-1721045534
groupId: g1
outerAdapterKey: mysql0

```

### CODE REVIEW

File: client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java [hunk 1]
```diff
@@ -151,8 +156,16 @@ private void initMappingConfig(String key, MappingConfig baseConfigMap, MirrorDb
...
         try (Connection conn = dataSource.getConnection(); Statement statement = conn.createStatement()) {
-            // 替换反引号
+
             String sql = ddl.getSql();
+            // 从配置中获取 targetDb，如果有则进行替换
+            String targetDb = mirrorDbConfig.getMappingConfig().getDbMapping().getTargetDb();
+            if (!StringUtils.isEmpty(targetDb)) {
+                String originalDbName = ddl.getDatabase();
+                String regex = String.format("\\b%s\\b([.`])", originalDbName);
+                sql = sql.replaceAll(regex, targetDb + "$1");
```
Comment (wenshao) [inline client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java:166]: **[Critical]** `originalDbName` is interpolated into the regex pattern without `Pattern.quote()`, and `targetDb` is used in the `replaceAll` replacement string without `Matcher.quoteReplacement()`. Regex metacharacters in either value cause misbehaviour or a crash. — Failure scenario: a source database named `my$db` produces the pattern `\bmy$b\b([.`])` where `$` is an end-of-line anchor — the pattern never matches, so DDL executes against the source db name on the target. A `targetDb` configured as `my$db` causes `Matcher` to interpret `$b` as a group reference, throwing `IllegalArgumentException: Illegal group reference`, crashing DDL sync.

```suggestion
                String regex = String.format("\\b%s\\b([.`])", Pattern.quote(originalDbName));
                sql = sql.replaceAll(regex, Matcher.quoteReplacement(targetDb) + "$1");
```

<details><summary>中文说明</summary>

`originalDbName` 直接拼入正则表达式而未使用 `Pattern.quote()` 转义，`targetDb` 直接拼入 `replaceAll` 的替换字符串而未使用 `Matcher.quoteReplacement()` 转义。如果数据库名包含正则元字符（如 `$`、`.`、`(`），会导致模式损坏或抛出异常。例如：源库名为 `my$db` 时，`$` 被解释为行尾锚点，正则永远无法匹配，DDL 将在目标端使用源库名执行；`targetDb` 配置为 `my$db` 时，`$b` 被解释为分组引用，抛出 `IllegalArgumentException`，导致 DDL 同步崩溃。

</details>

_— qwen3.8-max-preview via Qwen Code /review_

Comment (wenshao) [inline client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java:166]: **[Critical]** The regex replaces the database name in **all** positions where it appears followed by `.` or a backtick — including inside SQL string literals and non-qualifier backtick-quoted identifiers (columns, tables that share the database's name). This silently corrupts DDL content. — Failure scenario: source db `app`, targetDb `app_mirror`. DDL: `` CREATE TABLE `app`.`users` (`app` VARCHAR(50), path DEFAULT 'app.cfg') `` → the regex matches three times: the qualifier `app`.` (correct), the column `` `app` `` (closing backtick satisfies `[.`]` — column silently renamed to `app_mirror`), and the string literal `'app.cfg'` (default value corrupted). All three replacements execute successfully — the corruption is silent.

```suggestion
                // TODO: use a SQL-aware rewrite (e.g., split on single-quoted strings and
                // only rewrite outside them; match only `dbName`. qualifier patterns)
                String regex = String.format("\\b%s\\b(\\.)", Pattern.quote(originalDbName));
                sql = sql.replaceAll(regex, Matcher.quoteReplacement(targetDb) + "$1");
```

<details><summary>中文说明</summary>

该正则表达式会替换所有后跟 `.` 或反引号的数据库名出现位置——包括 SQL 字符串字面量内部和非限定符位置的反引号标识符（与库名同名的列名、表名）。这会导致 DDL 内容被静默篡改。例如：源库 `app`，目标库 `app_mirror`，DDL 为 `` CREATE TABLE `app`.`users` (`app` VARCHAR(50), path DEFAULT 'app.cfg') `` 时，正则匹配三次：限定符 `app`.`（正确）、列名 `` `app` ``（闭合反引号满足 `[.`]`，列名被静默改为 `app_mirror`）、字符串字面量 `'app.cfg'`（默认值被篡改）。所有替换均成功执行，损坏是静默的。建议改用 SQL 感知的重写方式，或至少将正则限制为仅匹配 `dbName.` 限定符模式。

</details>

_— qwen3.8-max-preview via Qwen Code /review_

File: client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java [hunk 2]
```diff
@@ -151,8 +156,16 @@ private void initMappingConfig(String key, MappingConfig baseConfigMap, MirrorDb
      */
     private void executeDdl(MirrorDbConfig mirrorDbConfig, Dml ddl) {
         try (Connection conn = dataSource.getConnection(); Statement statement = conn.createStatement()) {
-            // 替换反引号
+
             String sql = ddl.getSql();
+            // 从配置中获取 targetDb，如果有则进行替换
+            String targetDb = mirrorDbConfig.getMappingConfig().getDbMapping().getTargetDb();
```
Comment (wenshao) [inline client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java:162]: **[Suggestion]** `targetDb` from configuration is substituted into DDL SQL executed via `Statement.execute()` without validation as a legal SQL identifier. The DML path (`SyncUtil.getDbTableName`) wraps it in identifier quotes; the DDL path does not. — Concrete cost: a config value containing `;`, `--`, or quotes can corrupt the DDL syntax or (if `allowMultiQueries=true`) enable injection. Consider validating against `^[\\w$]+$` before use.

<details><summary>中文说明</summary>

配置中的 `targetDb` 未经 SQL 标识符合法性校验即被替换进通过 `Statement.execute()` 执行的 DDL SQL 中。DML 路径（`SyncUtil.getDbTableName`）会用反引号包裹，但 DDL 路径没有。如果配置值包含 `;`、`--` 或引号，可能破坏 DDL 语法，或在 `allowMultiQueries=true` 时导致注入。建议在使用前校验 `^[\\w$]+$`。

</details>

_— qwen3.8-max-preview via Qwen Code /review_

File: client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java [hunk 3]
```diff
@@ -151,8 +156,16 @@ private void initMappingConfig(String key, MappingConfig baseConfigMap, MirrorDb
...
+            if (!StringUtils.isEmpty(targetDb)) {
+                String originalDbName = ddl.getDatabase();
```
Comment (wenshao) [inline client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java:164]: **[Suggestion]** No test coverage for the new DDL rewriting logic or the `targetDb` resolution in `initMappingConfig`. No existing tests exercise `RdbMirrorDbSyncService` at all. — Concrete cost: the regex edge cases above (metacharacters, string literals, unqualified DDL) would not be caught before release. Consider adding unit tests for `executeDdl` covering: simple qualified DDL, db names with regex metacharacters, DDL with string literals containing the db name, and `targetDb` being null/empty.

<details><summary>中文说明</summary>

新增的 DDL 重写逻辑和 `initMappingConfig` 中的 `targetDb` 解析均无测试覆盖。现有测试中没有任何一个涉及 `RdbMirrorDbSyncService`。上述正则边界情况（元字符、字符串字面量、非限定 DDL）在发布前无法被发现。建议为 `executeDdl` 添加单元测试，覆盖：简单限定 DDL、包含正则元字符的库名、包含库名的字符串字面量的 DDL、以及 `targetDb` 为 null/空的情况。

</details>

_— qwen3.8-max-preview via Qwen Code /review_

File: client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java (hunk 1 above)
Comment (wenshao) [inline client-adapter/rdb/src/main/java/com/alibaba/otter/canal/client/adapter/rdb/service/RdbMirrorDbSyncService.java:166]: **[Suggestion]** When `targetDb` is configured, DDL whose SQL text does not contain the source database name (unqualified DDL, e.g. `CREATE TABLE mytable (id INT)` after `USE mydb`) is not rewritten. The connection has no `setCatalog()` call, so the DDL executes against the JDBC URL's default database while DML targets `targetDb` — a DML/DDL target split. — Concrete cost: subsequent DML for that table fails with "table doesn't exist" in `targetDb`. Consider calling `conn.setCatalog(targetDb)` before executing the DDL when `targetDb` is configured.

<details><summary>中文说明</summary>

当配置了 `targetDb` 时，SQL 文本中不包含源库名的 DDL（非限定 DDL，例如 `USE mydb` 后执行 `CREATE TABLE mytable (id INT)`）不会被重写。连接没有调用 `setCatalog()`，因此 DDL 会在 JDBC URL 的默认数据库上执行，而 DML 目标是 `targetDb`——导致 DML/DDL 目标不一致。后续对该表的 DML 会因 `targetDb` 中不存在该表而失败。建议在配置了 `targetDb` 时，执行 DDL 前调用 `conn.setCatalog(targetDb)`。

</details>

_— qwen3.8-max-preview via Qwen Code /review_

**Review by wenshao — CHANGES_REQUESTED** [review]
